main/display/display.c

```diff
@@ -300,21 +300,21 @@ static void rotatePixel(int16_t* x, int16_t* y, int16_t rotateDeg, int16_t width
         // (x, y) -> (y, -x)
         int16_t tmp = wx;
         wx = wy;
-        wy = -tmp;
+        wy = width - 1 - (tmp + (width / 2)) - (width / 2);
```

Comment:
This simplifies to `wy = -(1 + tmp)` I think

Reply:
![image](https://user-images.githubusercontent.com/231180/199867545-11a28075-4945-4aa1-a69f-f98177e7a15c.png)

Reply:
not quite, since technically with integer division it's `w - 1 - (t + floor(w / 2)) - floor(w / 2)`

![image](https://user-images.githubusercontent.com/666431/199871893-4b837cbf-1303-4b5f-b254-d35e5934ea93.png)

I can get combin ethe `width / 2`s though

Reply:
https://godbolt.org/ says `wy = -2 * (width / 2) - tmp + width - 1;` is 21 instructions and `wy = width - 1 - (tmp + (width / 2)) - (width / 2);` is 25 instructions, so yeah, let's optimize! for per-pixel ops, this does matter.

Reply:
what if we move `-(width / 2)` to a var and change that to `wy = tx - tmp + width - 1`? If I used the right settings (I probably did not) that drops the instruction count way down

Reply:
plug it into https://godbolt.org/, select the ESP32-S2 gcc 8.4.0 compiler and see. I've been using
```C
#include <stdint.h>

void mathA(void)
{
    int16_t wx, wy;
    wx = -(1 + wy);
}

void mathB(void)
{
    int16_t wy, width, tmp;
    wy = -2 * (width / 2) - tmp + width - 1;
}
```

Reply:
I'll wait for your results before merging.

Reply:
Oh yeah, looks like that drops it down to 12 instructions

Reply:
Wait hang on, I might have screwed that up

Reply:
It's still 21 instructions but now we get to reuse a bunch of them for the translation, so I think that still makes it slightly better overall at least

Reply:
I'll take that optimization!